Packages/com.unity.inputsystem/InputSystem/Controls/InputControlList.cs

```diff
@@ -348,15 +348,29 @@ public void CopyTo(TControl[] array, int arrayIndex)
 
         public int IndexOf(TControl item)
         {
+            return IndexOf(item, 0);
+        }
+
+        public int IndexOf(TControl item, int startIndex, int count = -1)
+        {
+            if (startIndex < 0)
+                throw new ArgumentOutOfRangeException(nameof(startIndex), "startIndex cannot be negative");
+
             if (m_Count == 0)
                 return -1;
 
             var index = ToIndex(item);
             var indices = (ulong*)m_Indices.GetUnsafeReadOnlyPtr();
 
-            for (var i = 0; i < m_Count; ++i)
-                if (indices[i] == index)
-                    return i;
+            if (count < 0)
+                count = Mathf.Max(m_Count - startIndex, 0);
+
+            if (startIndex + count > m_Count)
```

Comment:
Nitpick, can be ignored, but in general it would be great to avoid doing work like `var index = ToIndex(item);` and `var indices = (ulong*)m_Indices.GetUnsafeReadOnlyPtr();` before we're done with all "early outs" like `if (startIndex + count > m_Count)`

Reply:
Good catch. Fixed.